mappings/clacso.py

```diff
@@ -1,23 +1,119 @@
-from .json import JSONMapping
+import re
+import requests
+from bs4 import BeautifulSoup
 
-class CLACSOMapping(JSONMapping):
-    def __init__(self, source):
-        super().__init__(source, {})
+from mappings.xml import XMLMapping
+
+class CLACSOMapping(XMLMapping):
+    def __init__(self, source, namespace, constants):
+        super(CLACSOMapping, self).__init__(source, namespace, constants)
         self.mapping = self.createMapping()
     
     def createMapping(self):
         return {
-            'title': ('title', '{0}'),
-            'authors': ('authors', '{0}|||true'),
-            'abstract': ('description', '{0}'),
-            'dates': [('publication_date', '{0}|publication_date')],
-            'identifiers': 
-                [('isbn', '{0}|isbn')]
-            ,
-            'spatial': ('edition_location', '{0}'),
-            'has_part': [('url', '1|{0}|clacso|application/pdf|{{"catalog": false, "download": true, "reader": false, "embed": false}}')] 
+            'title': ('./dc:title/text()', '{0}'),
+            'authors': [('./dc:creator/text()', '{0}|||true')],
+            'abstract': ('./dc:description/text()', '{0}'),
+            'dates': [
+                (
+                    [
+                        './dc:date/text()',
+                    ],
+                    '{0}'
+                )
+            ],
+            'identifiers': [
+                ('./dc:identifier/text()', '{0}'),
+            ],
+            'medium' : [
+                ('./dc:type/text()', '{0}')
+            ],
+            'has_part': [(
+                './dc:identifier/text()',
+                '1|{0}|clacso|text/html|{{"reader": false, "download": false, "catalog": false, "embed": true}}'
+            )]
         }
 
     def applyFormatting(self):
         self.record.source = 'clacso'
-        self.record.source_id = f'{self.record.source}'
+        handle_url = "https://biblioteca-repositorio.clacso.edu.ar/handle/CLACSO"
+        self.formatting_source_id(handle_url)
```

Comment:
recommend changing these function names to `format_`. using the infinitive of the verb rather than the gerund I think is more explicit and shows action.